src/DynamoUtilities/CertificateVerification.cs

```diff
@@ -3,12 +3,194 @@
+            IntPtr StateData = IntPtr.Zero;
+            String URLReference = null;
+            WinTrustDataProvFlags ProvFlags = WinTrustDataProvFlags.RevocationCheckChainExcludeRoot;
+            WinTrustDataUIContext UIContext = WinTrustDataUIContext.Execute;
+
+            // constructor for silent WinTrustDataChoice.File check
+            public WinTrustData(WinTrustFileInfo _fileInfo)
+            {
+                // On Win7SP1+, don't allow MD2 or MD4 signatures
+                if ((Environment.OSVersion.Version.Major > 6) ||
+                    ((Environment.OSVersion.Version.Major == 6) && (Environment.OSVersion.Version.Minor > 1)) ||
+                    ((Environment.OSVersion.Version.Major == 6) && (Environment.OSVersion.Version.Minor == 1) && !String.IsNullOrEmpty(Environment.OSVersion.ServicePack)))
+                {
+                   ProvFlags |= WinTrustDataProvFlags.DisableMD2andMD4;
+                }
+
+                WinTrustFileInfo wtfiData = _fileInfo;
+                FileInfoPtr = Marshal.AllocCoTaskMem(Marshal.SizeOf(typeof(WinTrustFileInfo)));
+                Marshal.StructureToPtr(wtfiData, FileInfoPtr, false);
+            }
+            public void Dispose()
+            {
+                if(FileInfoPtr != IntPtr.Zero) {
+                    Marshal.FreeCoTaskMem(FileInfoPtr);
+                    FileInfoPtr = IntPtr.Zero;
+                }
+            }
+        }
+        #endregion
+
+        enum WinVerifyTrustResult : uint
+        {
+            Success = 0,
+            ProviderUnknown = 0x800b0001,           // Trust provider is not recognized on this system
+            ActionUnknown = 0x800b0002,         // Trust provider does not support the specified action
+            SubjectFormUnknown = 0x800b0003,        // Trust provider does not support the form specified for the subject
+            SubjectNotTrusted = 0x800b0004,         // Subject failed the specified verification action
+            FileNotSigned = 0x800B0100,         // TRUST_E_NOSIGNATURE - File was not signed
+            SubjectExplicitlyDistrusted = 0x800B0111,   // Signer's certificate is in the Untrusted Publishers store
+            SignatureOrFileCorrupt = 0x80096010,    // TRUST_E_BAD_DIGEST - file was probably corrupt
+            SubjectCertExpired = 0x800B0101,        // CERT_E_EXPIRED - Signer's certificate was expired
+            SubjectCertificateRevoked = 0x800B010C,     // CERT_E_REVOKED Subject's certificate was revoked
+            UntrustedRoot = 0x800B0109          // CERT_E_UNTRUSTEDROOT - A certification chain processed correctly but terminated in a root certificate that is not trusted by the trust provider.
+        }
+
+        public class WinTrust
+        {
+            private static readonly IntPtr INVALID_HANDLE_VALUE = new IntPtr(-1);
+            // GUID of the action to perform
+            private const string WINTRUST_ACTION_GENERIC_VERIFY_V2 = "{00AAC56B-CD44-11d0-8CC2-00C04FC295EE}";
```

Comment:
should this be static?

Reply:
const is set at compile time vs static readonly set at runtime